library/core/src/intrinsics.rs

```diff
@@ -2828,3 +2839,16 @@ pub const unsafe fn transmute_unchecked<Src, Dst>(src: Src) -> Dst {
     // SAFETY: It's a transmute -- the caller promised it's fine.
     unsafe { transmute_copy(&ManuallyDrop::new(src)) }
 }
+
+/// Polyfill for bootstrap
+#[cfg(bootstrap)]
+pub const unsafe fn write_via_move<T>(ptr: *mut T, value: T) {
+    use crate::mem::*;
+    // SAFETY: the caller must guarantee that `dst` is valid for writes.
+    // `dst` cannot overlap `src` because the caller has mutable access
+    // to `dst` while `src` is owned by this function.
+    unsafe {
+        copy_nonoverlapping::<T>(&value, ptr, 1);
+        forget(value);
```

Comment:
Not important, but could you use `ManuallyDrop` instead?

Reply:
I probably could, but the preexisting implementation of `write` went out of its way to use `intrinsics::forget`

https://github.com/rust-lang/rust/blob/4b87ed9d0f18166a26aadb5289bac004e058b90f/library/core/src/ptr/mod.rs#L1369

even though `mem::forget` doesn't

https://github.com/rust-lang/rust/blob/4b87ed9d0f18166a26aadb5289bac004e058b90f/library/core/src/mem/mod.rs#L148-L150

which is *probably* just because it has much better codegen, but in case it actually matters I'd rather just leave it like this since it sounds like you don't feel particularly strongly about it.

Reply:
Okay 👍🏻